File: Tests/IssuesTests.go
package GoldLibrary.Tests.Shared

import (
	"fmt"
)

type span struct {
	start, end int
}

func DoAppendArrayTest() bool {
	var pendingAttr   [2]span
	var attr          [][2]span
	pendingAttr[0].start = 1
	pendingAttr[0].end = 2
	pendingAttr[1].start = 3
	pendingAttr[1].end = 4

	attr = append(attr, pendingAttr)
	attr[0][0].start = 200;

	return pendingAttr[0].start == 1
}

File: Tests/IssuesTests.pas
namespace GoldLibrary.Tests.Shared;

uses
  RemObjects.Elements.EUnit,
  RemObjects.Elements.RTL;

type
  AppendArrayTests = public class(Test)
  public
    method AppendArrayTest;
    begin
      Assert.IsTrue(TestApplication2.DoAppendArrayTest());
    end;
  end;



end.